packages/block-editor/src/store/reducer.js

```diff
@@ -1873,6 +1873,43 @@ export function highlightedBlock( state, action ) {
 	return state;
 }
 
+/**
+ * Reducer returning current spotlighted block.
+ *
+ * @param {string|null} state  Current clientId or null.
+ * @param {Object}      action Dispatched action.
+ *
+ * @return {string|null} Updated state.
+ */
+export function hasBlockSpotlight( state, action ) {
+	switch ( action.type ) {
+		case 'TOGGLE_BLOCK_SPOTLIGHT':
+			const { clientId, hasBlockSpotlight: _hasBlockSpotlight } = action;
+
+			if ( _hasBlockSpotlight ) {
+				return clientId;
+			} else if ( state === clientId ) {
+				return null;
+			}
+			return state;
+		case 'SELECT_BLOCK':
+			if ( action.clientId !== state ) {
+				return null;
+			}
+			return state;
+		case 'CLEAR_SELECTED_BLOCK':
+			return null;
+		case 'SELECTION_CHANGE': {
+			if ( ! action.clientId && ! action.start && ! action.end ) {
+				return null;
+			}
+			return state;
+		}
```

Comment:
Question (non-blocking): This seems to be different from the highlight block reducer. What are these resolving?

Reply:
As I understand it, blocks are only spotlighted when a comment thread is clicked. I think the spotlight state needs to be disabled at all other times. Without these two reducers, the spotlight will not be removed when clicking on anything other than the block.

https://github.com/user-attachments/assets/1394df98-a05b-4b03-9d78-29d02df92719

A similar issue occurs with `toggleBlockHighlight`, so we might need a similar reducer for it

https://github.com/user-attachments/assets/87a0bbce-b818-4380-9947-9a76006a5f6c